scripts/real_project_tests.py

```diff
@@ -0,0 +1,113 @@
+import os
+import re
+import subprocess
+import time
+from collections import defaultdict
+
+import click
+
+from common import git_command
+
+ROOT_PATH = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
+TEST_DATA_PATH = os.path.join(ROOT_PATH, "testData")
+
+PROJECTS = [
+    ("cargo", "https://github.com/rust-lang/cargo"),
+    ("mysql_async", "https://github.com/blackbeam/mysql_async"),
+    ("tokio", "https://github.com/tokio-rs/tokio"),
+    ("amethyst", "https://github.com/amethyst/amethyst"),
+    ("clap", "https://github.com/clap-rs/clap"),
+    ("diesel", "https://github.com/diesel-rs/diesel"),
+    ("rust-analyzer", "https://github.com/rust-analyzer/rust-analyzer"),
+    ("xi-editor", "https://github.com/xi-editor/xi-editor"),
+    ("juniper", "https://github.com/graphql-rust/juniper")
+]
+
+
+@click.command("init")
+def init():
+    for (project, git_url) in PROJECTS:
+        directory = os.path.join(TEST_DATA_PATH, project)
+        if not os.path.isdir(directory):
+            print("Cloning {}".format(directory))
+            subprocess.run(["git", "clone", git_url], cwd=TEST_DATA_PATH, check=True)
```

Comment:
Hey, let's add the `--depth 1` parameter here to reduce the cloning time of the project library just for testing.

Reply:
Good point. I initially thought that the tests are testing a fixed revision, but then I realized that they are testing master, so this should be fine.